storm-client/src/jvm/org/apache/storm/StormSubmitter.java

```diff
@@ -235,7 +237,14 @@ public static void submitTopologyAs(String name, Map<String, Object> topoConf, S
         conf.putAll(topoConf);
         topoConf.putAll(prepareZookeeperAuthentication(conf));
 
-        validateConfs(conf, topology, name);
+        validateConfs(conf);
+
+        try {
+            StormCommon.validateCycleFree(topology, name);
+        } catch (InvalidTopologyException ex) {
+            LOG.warn(ex.get_msg(), ex);
+            System.out.println(ex.get_msg());
```

Comment:
We should remove this

Reply:
Did we not want to warn the user when submitting?

Reply:
`LOG.warn` should be sufficient. We don't use System.out.println()

Reply:
ok. removed sysout.